apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1377,9 +1377,7 @@ func (u *UniterAPI) oneEnterScope(ctx context.Context, canAccess common.AuthFunc
 		relUUID,
 		unitName,
 		settings,
-		// TODO(aflynn): pass in a create subordinate application function from
-		// the application service here
-		subordinateCreator(nil),
+		subordinateCreator(u.applicationService.AddSubordinateUnit),
```

Comment:
`subordinateCreator` does no work here. Remove it and just pass `u.applicationService.AddSubordinateUnit`

Reply:
The type system doesn't allow it. This is an interface in the relation domain, and it needs something that it can call the method on. If there is a more concise way to do it I'll switch to it, but this was the best I could find that worked.